core/src/main/java/org/hisp/dhis/android/core/category/CategoryCategoryOptionLinkStoreImpl.java

```diff
@@ -4,47 +4,126 @@
+import java.util.List;
+
 
 public class CategoryCategoryOptionLinkStoreImpl implements CategoryCategoryOptionLinkStore {
 
     private final DatabaseAdapter databaseAdapter;
     private final SQLiteStatement insertStatement;
+    private final SQLiteStatement deleteStatement;
+    private final SQLiteStatement updateStatement;
 
     private static final String INSERT_STATEMENT =
             "INSERT INTO " + CategoryCategoryOptionLinkModel.TABLE + " (" +
                     CategoryCategoryOptionLinkModel.Columns.CATEGORY + ", " +
                     CategoryCategoryOptionLinkModel.Columns.CATEGORY_OPTION + ") " +
                     "VALUES(?, ?);";
 
+    private static final String DELETE_STATEMENT =
+            "DELETE FROM " + CategoryCategoryOptionLinkModel.TABLE +
+                    " WHERE " + CategoryCategoryOptionLinkModel.Columns.CATEGORY + " =?" + " AND "
+                    + CategoryCategoryOptionLinkModel.Columns.CATEGORY_OPTION + "=?;";
+
+    private static final String UPDATE_STATEMENT =
+            "UPDATE " + CategoryCategoryOptionLinkModel.TABLE + " SET " +
+                    CategoryCategoryOptionLinkModel.Columns.CATEGORY + " =?, " +
+                    CategoryCategoryOptionLinkModel.Columns.CATEGORY_OPTION + " =?" +
+                    " WHERE " + CategoryCategoryOptionLinkModel.Columns.CATEGORY + " =? AND " +
+                    CategoryCategoryOptionLinkModel.Columns.CATEGORY_OPTION + " =?;";
+
+    private static final String FIELDS =
+            CategoryCategoryOptionLinkModel.TABLE + "."
+                    + CategoryCategoryOptionLinkModel.Columns.CATEGORY + "," +
+                    CategoryCategoryOptionLinkModel.TABLE + "."
+                    + CategoryCategoryOptionLinkModel.Columns.CATEGORY_OPTION;
+
+
+    private static final String QUERY_ALL_CATEGORY_OPTION_LINKS = "SELECT " +
+            FIELDS + " FROM " + CategoryCategoryOptionLinkModel.TABLE;
 
     public CategoryCategoryOptionLinkStoreImpl(DatabaseAdapter databaseAdapter) {
         this.databaseAdapter = databaseAdapter;
         this.insertStatement = databaseAdapter.compileStatement(INSERT_STATEMENT);
+        this.deleteStatement = databaseAdapter.compileStatement(DELETE_STATEMENT);
+        this.updateStatement = databaseAdapter.compileStatement(UPDATE_STATEMENT);
     }
 
     @Override
-    public long insert(@NonNull CategoryCategoryOptionLinkModel entity) {
+    public long insert(@NonNull CategoryCategoryOptionLinkModel categoryOptionLinkModel) {
 
-        validate(entity);
+        validate(categoryOptionLinkModel);
 
-        bind(insertStatement, entity);
+        bind(insertStatement, categoryOptionLinkModel);
 
         return executeInsert();
     }
 
+    @Override
+    public int delete() {
+        return databaseAdapter.delete(CategoryCategoryOptionLinkModel.TABLE);
+    }
+
+    @Override
+    public boolean delete(@NonNull CategoryCategoryOptionLinkModel categoryOptionLinkModel) {
```

Comment:
In all stores in this method that delete by uid returns a int type, please can you refactor it?

Reply:
I changed the return type to an int but this entity is a link table and it has two primary keys if I remove the object I will have to have two uids instead of one object.

Do you have that I use two uids instead of one object?